### PR TITLE
Add simple-react-clipboard

### PR DESCRIPTION
**Pull request by huytranvn**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
## Formatting

The following files are not formatted:
1. types/simple-react-clipboard/simple-react-clipboard-tests.tsx

Consider running `pnpm dprint fmt` on these files to make review easier.
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 487b62cbacac5fa94b4a6d96f44e16e9920dde6e
</p>

File: types/simple-react-clipboard/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { ReactNode } from "react";
+
+interface ClipboardProps {
+    render: () => ReactNode;
+    text: string;
+    props: object;
```
Comment (peterblazejewicz) [inline types/simple-react-clipboard/index.d.ts:6]:
```suggestion
    props?: object | undefined;
```
is this one optional as the others?
https://github.com/donfour/simple-react-clipboard/blob/master/src/lib/index.js#L64

Comment (huytranvn) [reply types/simple-react-clipboard/index.d.ts:6]: Yes, it should be optional. Nice catch ⭐

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@huytranvn thanks!

**Comment by huytranvn** [comment]
Ready to merge

**Pull request merged**